### PR TITLE
aded_mongoDB

### PR DESCRIPTION
**Pull request by Yuliia9800**

### CODE REVIEW

File: src/routes/movies.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+ *           type: string
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/Movie'
```
Comment (vasylkivskyy) [inline src/routes/movies.ts:194]: Receive wrong succesfull payload response in movies in SwaggerUI: [
  "string"
]

File: src/index.ts [hunk 2]
```diff
@@ -33,6 +47,14 @@ app.use((_err: Error, _req: Request, res: Response, _next: NextFunction) => {
   res.status(500).json({ error: 'Internal Server Error' });
 });
 
-app.listen(PORT, () => {
-  console.log(`Server listening on port ${PORT}`);
-});
+mongoose
+  .connect('mongodb+srv://uliasumik:3GyGBvHtpl4AyYjR@yuliia.pniib08.mongodb.net/?retryWrites=true&w=majority')
```
Comment (vasylkivskyy) [inline src/index.ts]: Don't push sensitive data to GitHub, use dotenv for example

File: src/routes/genres.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ *           type: string
+ *     responses:
+ *       200:
+ *         description: Genre deleted successfully
+ *       404:
+ *         description: Genre not found
+ *       500:
+ *         description: Internal server error
+ */
+router.delete('/:id', async (req: Request, res: Response) => {
```
Comment (vasylkivskyy) [inline src/routes/genres.ts:58]: Better to add validation middleware for movies and genres to check field and return 400 (not 500) error in response

File: src/routes/genres.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+ *                 description: The name of the genre.
+ *     responses:
+ *       200:
+ *         description: Genre created successfully
+ *       500:
+ *         description: Internal server error
+ */
+router.post('/', async (req: Request, res: Response) => {
+  try {
+    const genre = await Genre.create(req.body);
```
Comment (vasylkivskyy) [inline src/routes/genres.ts:113]: You can add checking if genre is existing

**Review by vasylkivskyy — CHANGES_REQUESTED** [review]

File: src/routes/movies.ts [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+import { Movie } from '../models/movie.model';
+import { moviesDataValidation } from '../validations/movies.validation';
+import { validationResult } from 'express-validator';
+
+const router = Router();
+
+/**
+ * @swagger
+ * /movies:
+ *   get:
```
Comment (vasylkivskyy) [inline src/routes/movies.ts:11]: Resolver error at paths./movies.get.responses.200.content.application/json.schema.items.$ref
Could not resolve reference: Could not resolve pointer: /components/schemas/Movie does not exist in document

Comment (vasylkivskyy) [reply src/routes/movies.ts:11]: in SwaggerUI

Comment (Yuliia9800) [reply src/routes/movies.ts:11]: Updated, could you please re-check

**Review by vasylkivskyy — APPROVED** [review]

**Pull request merged**